client/src/screens/search-groups.screen.js

```diff
@@ -68,21 +67,36 @@ class Group extends Component {
   constructor(props) {
     super(props);
     this.myGroups = this.props.myGroups;
-    this.joinGroupQuery = this.props.joinGroupQuery.bind(this);
-    this.joinGroup = this.props.joinGroup.bind(this, this.props.group);
+    this.joinGroup = this.joinGroup.bind(this);
+    this.groupUpdate = this.props.groupUpdate.bind(this);
+  }
+
+  joinGroup() {
+    const { groupUpdate, group } = this.props;
+    groupUpdate({ groupId: group.id }).then(() => {
+      // dont need to do anything, but we might in the future
+    }).catch((error) => {
+      Alert.alert(
+        'Error Joining Group',
+        error.message,
+        [
+          { text: 'OK', onPress: () => {} },
+        ],
+      );
+    });
   }
 
+
   render() {
     const { id, name } = this.props.group;
     const tags = this.props.group.tags.map(elem => `#${elem.name}`).join(',');
-
-    // color the already subscribed groups green
-    const memberAlready = _.some(this.myGroups, g => g.id === id);
-
+    this.state = {
```

Comment:
hmm I don't think you should store this in state.. why didn't what we had previous work? (using a local)
If you're going to go and calculate the state every time you render and the only place you use it is the render function it seems silly to store it in state.